src/plugins/dashboard/public/application/dashboard_app_controller.tsx

```diff
@@ -93,6 +93,11 @@ export interface DashboardAppControllerDependencies extends RenderDeps {
   navigation: NavigationStart;
 }
 
+export interface DashboardIncomingState {
+  type: string;
+  id: string;
```

Comment:
I'd add some comments on what these parameters mean.  is `id` always going to be a saved object id? Eventually I think we talked about replacing it with `input`? then the shape might be `Record<string, unknown>`.

Reply:
I've created a new service to resolve some of the issues around the typing of the state objects. Ideally, the shape of the state will be owned by a third plugin, in this case the embeddable plugin. I found that passing a type and a type guard to the `incoming` method was the best way to ensure the correct shape of the state due to the `redirectToApp` api's state being of type unknown.

Now, when we change the EmbeddablePackage type's shape to include `explicitInput` instead of `id`, we will get type typescript warnings everywhere that uses it!